designs/osc_high_availability/high_availability.md

```diff
@@ -0,0 +1,100 @@
+
+We have explored three approaches.
+1. OSC node cluster with Active/Passive setup using Pacemaker, Corosync and Distributed Replicated Block Device(DRBD).
+2. Segregate Application and Persistence Layers(The option has discarded since H2 does not provide the option of Active/Passive Cluster)
+3. Using NFS and Custom Monitor Script
+
+## OSC HA with Active/Passive Setup 
+OSC node cluster with Active/Passive setup using Pacemaker, Corosync and Distributed Replicated Block Device(DRBD).
+Active Node serves the requests where as Passive Node is Standby. So, on fails of Active Node, Passive Node will get promoted to Active Node.
+Pacemaker and Corosync maintains Service HA where It achieves maximum availability for our cluster services by detecting and recovering from node- and resource-level failures by making use of the messaging and membership capabilities provided by Corosync.
+DRBD is a software-based, shared-nothing, replicated storage solution mirroring 
+the content of block devices (hard disks, partitions, logical volumes etc.) between hosts.
+
+#### Active/Passive Architecture Diagram
+
+![](./images/OSC-HA-Using-DRBD.png) 
+
+**Pacemaker and Corosync provide Service HA where as DRBD data replication through Synchronous updates.**
+
+#### DRBD Architecture Diagram
+
+![](./images/DRBD-Arch-Diagram.png) 
+
+**DRBD is the replacement of shared storage systems by networked mirroring. Also, called Network based RAID 1 mirroring for the data. It allows read writes on Active Node where as completely disallows to Passive Node.**
+
+## OSC HA Through Segregating Application and Persistence Layers
+Currently, OSC is tightly coupled with the Application and Persistence Layers. So, creating HA for persistence layer is little hard. Seprating the Application and Persistence Layers will make the OSC loosly coupled. Also, creating the HA for both the layers will be easier and align with other openstack services.
+
+#### HA Architecture Diagram for Application and Persistence Layers
+
+![](./images/OSC-HA-Through-Segregation.png) 
+
+**Application and Persistence Layers seprated out with seprate HA. H2 Database has H2 Plugins for Active/Standby setup options and replications.**
+
+###### Note: Seprating Database is not feasible in OSC since H2 does not provide the Active/Passive Cluster option.Hence,discarding this option.
+
+## OSC HA Using NFS and Custom Monitoring Script
+NFS Hosts the Docker Volume File System and will be Mounted/Un-Mounted in the case Active/Passive. Custom Monitoring Script will Mount/Un-Mount the volume and Start the OSC Docker Container pointing to NFS File System. Pacemaker will execute the Custom Monitoring Script based on the status change.
+
+#### HA Architecture Diagram Using NFS and Custom Monitoring Script
```

Comment:
please add more details on where the script exists, how to configure pacemaker/corosync to execute script etc.

reference to their documentation would be nice.